### PR TITLE
NakamaClient now auto-refresh sessions using refresh_token.

### PR DESCRIPTION
**Pull request by Faless**
This is enabled by default, but can be turned off via the `client.auto_refresh` option.

The session is refreshed if it would expire in `client.auto_refresh_seconds` seconds (300 = 5min by default).

This PR depends on #73 .

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ The format is based on [keep a changelog](http://keepachangelog.com/) and this p
 - Add purchase validation functions.
 - Add Apple authentication functions.
 - Add "demote_group_users_async" function.
+- A session can be refreshed on demand with "session_refresh_async" method.
+- Session and/or refresh tokens can now be disabled with a client logout.
```
Comment (lugehorsam) [inline CHANGELOG.md:15]: I think "invalidated" would be a better word here than disabled.

**Comment by lugehorsam** [comment]
Hey @Faless, there are some differences between this design and the .NET one that I wanted to document so we can compare and contrast approaches. Yours is simpler which I like, so it will be a weighing of the simplicity with some extra functionality in the .NET one.

In the .NET PR
- User can assign callback to refresh attempts in order to update UI or run any other arbitrary logic.
- User can configure refresh policy on a request-by-request basis
- Session refresh is happening in the the client wrapper rather than the generated core (doesn't affect user so this is minor and can be changed as we need.)

Once the team discusses the merits of each, we can circle back to this PR.

**Comment by Faless** [comment]
> User can assign callback to refresh attempts in order to update UI or run any other arbitrary logic.

That's actually a very good idea, I'll implement it. :+1: 

> User can configure refresh policy on a request-by-request basis

I don't really see use cases for having per-request logic and in those rare cases one could simply use a different client instance with the same session.

Something like this would also work 99% of the time, but likely using a different client instance is more straight forward.

```gdscript
client.auto_refresh = false
var req = client.my_request_async()
client.auto_refresh = true
var result = yield(req, "completed")
```

Again, this is assuming those are rare circumstances, which seems the case to me, but I might be missing something.

File: CHANGELOG.md (hunk 1 above)
Comment (Faless) [reply CHANGELOG.md:15]: Sorry, I just realized I didn't address this before merging... I forgot to force push my local branch :disappointed:

**Pull request merged**